packages/web3torrent/src/components/domain-budget-table/DomainBudgetTable.tsx

```diff
@@ -52,71 +42,53 @@ export const DomainBudgetTable: React.FC<DomainBudgetTableProps> = props => {
 
   const receiveBudget = bigNumberify(budgetCache.budgets[0].availableReceiveCapacity);
 
-  const inverseSpentFraction = spent.gt(0) ? spendBudget.div(spent).toNumber() : undefined;
-  const spentFraction = inverseSpentFraction ? 1 / inverseSpentFraction : 0;
-  const inverseRecievedFraction = received.gt(0)
-    ? receiveBudget.div(received).toNumber()
-    : undefined;
-  const receiveFraction = inverseRecievedFraction ? 1 / inverseRecievedFraction : 0;
+  const total = spendBudget
+    .add(receiveBudget)
+    .add(spent)
+    .add(received); // this should be constant
+
+  const spendBudgetPercentage = spendBudget.gt(0) ? 100 / total.div(spendBudget).toNumber() : 0;
```

Comment:
This seems better to me.

Reply:
Note slightly contorted logic around computing percentages, because we use ethers BigNumbers which are always integers. This could probably be factored out into a helper function.